sockets.js

```diff
@@ -443,48 +471,55 @@ if (cluster.isMaster) {
 		sockets.clear();
 		channels.clear();
 		subchannels.clear();
+
 		app.close();
 		if (appssl) appssl.close();
+
+		// Let the server(s) finish closing.
+		setImmediate(() => process.exit(0));
 	});
 
 	// this is global so it can be hotpatched if necessary
 	let isTrustedProxyIp = Dnsbl.checker(Config.proxyip);
 	let socketCounter = 0;
 	server.on('connection', socket => {
-		if (!socket) {
-			// For reasons that are not entirely clear, SockJS sometimes triggers
-			// this event with a null `socket` argument.
-			return;
-		} else if (!socket.remoteAddress) {
-			// This condition occurs several times per day. It may be a SockJS bug.
+		// For reasons that are not entirely clear, SockJS sometimes triggers
+		// this event with a null `socket` argument.
+		if (!socket) return;
+
+		if (!socket.remoteAddress) {
+			// SockJS will sometimes fail to cache a connection's IP, protocol,
+			// and address sometimes. When this happens is likely dependent on
+			// the transport being used and what browser's running the client.
+			// TODO: look more into why this happens.
 			try {
 				socket.destroy();
 			} catch (e) {}
 			return;
 		}
 
-		let socketid = socket.id = '' + (++socketCounter);
+		let socketid = '' + (++socketCounter);
 		sockets.set(socketid, socket);
 
-		if (isTrustedProxyIp(socket.remoteAddress)) {
+		let socketip = socket.remoteAddress;
+		if (isTrustedProxyIp(socketip)) {
 			let ips = (socket.headers['x-forwarded-for'] || '').split(',');
-			let ip;
-			while ((ip = ips.pop())) {
-				ip = ip.trim();
-				if (!isTrustedProxyIp(ip)) {
-					socket.remoteAddress = ip;
+			for (let i = ips.length; i--;) {
+				let proxy = ips[i].trim();
+				if (!isTrustedProxyIp(proxy)) {
+					socketip = proxy;
 					break;
 				}
 			}
 		}
 
-		process.send(`*${socketid}\n${socket.remoteAddress}\n${socket.protocol}`);
+		process.send(`*${socketid}\n${socketip}\n${socket.protocol}`);
 
 		socket.on('data', message => {
 			// drop empty messages (DDoS?)
 			if (!message) return;
 			// drop messages over 100KB
-			if (message.length > 100000) {
+			if (message.length > (100 * 1024)) {
```

Comment:
Also unneeded. Note that ``message.length`` doesn't even equal the byte size of the string.
Eg.

```js
'\u{1f000}'.length // 2
Buffer.byteLength('\u{1f000}') // 2
```

Reply:
This is more me nitpicking over kilobytes vs kibibytes, not so much to do with string length not always being accurate when Unicode gets involved.

Reply:
But yeah, the change isn't exactly needed

Reply:
I'm fine with this change being here.